js-mentoring-program-master/tasks/async/async.js

```diff
@@ -4,48 +4,71 @@ const {
     callbackFunctionError,
     promiseFunctionError
 } = require("./async_api");
-
+ 
 /**
  * console.log data got from callbackFunction
  */
 function printDataCallback() {
+    const cb = (str, data) => console.log(data);
+    callbackFunction(cb);
 }
-
+ 
 /**
  * console.log data got from promiseFunction
  */
 function printDataPromise() {
+    return promiseFunction()
+        .then(value => console.log(value));
 }
-
+ 
 /**
  * console.log data got from promiseFunction using async/await
  */
 async function printDataAsyncAwait() {
+    const data = await promiseFunction();
+    console.log(data);
 }
-
+ 
 /**
  * throw error data from callbackFunctionError
  */
 function handleErrorCallback() {
+    const cb = (error) => {
+        throw error;
+    };
+    callbackFunctionError(cb);
 }
-
+ 
 /**
  * throw error come from promiseFunctionError
  */
+ 
 function handlePromiseError() {
+    try {
```

Comment:
Try-catch blocks should not be used here. You have to figure out which method of https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Promise can be applied to handle errors.